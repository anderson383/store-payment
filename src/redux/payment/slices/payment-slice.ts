import { createSlice } from "@reduxjs/toolkit"
import { setModalState } from '../actions/payment.action'
import { PaymentModel } from "../models/payment.models";

const INITIAL_STATE: PaymentModel = {
  modalPayment: false
}

export const paymentSlice = createSlice({
  name: 'payment',
  initialState: INITIAL_STATE,
  reducers: {
    setModalState,
  }
})

export const {
  setModalState: setModalStateSlice,
} = paymentSlice.actions;
