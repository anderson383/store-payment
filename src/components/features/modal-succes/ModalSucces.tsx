import Modal from "../../../components/ui/molecules/modal/modal"
import styles from './ModalSuccess.module.scss'
import { useDispatch, useSelector } from "react-redux"
import { RootStateType } from "redux/store"
import { setModalSuccessSlice } from "../../../redux/payment/slices/payment-slice"
import { useNavigate } from "react-router-dom"
import { Button } from "../../../components/ui/atoms/button/Button"


export const ModalSuccess = () => {
  const dispatch = useDispatch()
  const navigate = useNavigate()
  const { modalSuccess } = useSelector(({ paymentReducer }: RootStateType) => paymentReducer.payment)

  const handleModal = () => {
    dispatch(setModalSuccessSlice(!modalSuccess))
  }

  const redirect = () => {
    handleModal()
    navigate('/')
  }
  return (
    <Modal
      openModal={modalSuccess} 
      setOpenModal={() => handleModal()}
      closeOutClick={true}
      title={'Payment Succesful'}
    >
      <div className={styles.styles}>
        <p>You will receive your order in the coming days. <br /> <br /> Thank you for your purchase.</p>
      </div>

      <div className={styles.action}>
        <Button size="fullwidth"  onClick={() => redirect()} text={'Ok'} />
      </div>
    </Modal>
  )
}