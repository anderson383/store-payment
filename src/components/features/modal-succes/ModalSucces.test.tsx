
import {
  fireEvent,
  render,
  screen
} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { act } from 'react';
import { ModalSuccess } from './ModalSucces';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { createRoot } from 'react-dom/client';
import { setModalSuccessSlice } from '../../../redux/payment/slices/payment-slice';

const mockStore:any = configureStore([]);

describe('Tests in ModalSuccess component', () => {

  let store:any;

  
  let container:any = null;
  let root:any = null;

  afterEach(() => {
    act(() => {
      root.unmount();
    })
    container?.remove();
    container = null;
  });

  beforeEach(() => {
    container = document.createElement('div');
    container.setAttribute('id', 'radix-:r0:');
    container.setAttribute('role', 'dialog');
    root = createRoot(container!);

    document.body.appendChild(container);
  });

  
  beforeEach(() => {
    store = mockStore({
      paymentReducer: {
        payment: {
          modalSuccess: true, // Estado inicial del modal
        },
      },
    });
  });

  const setup = () => {
    return render(
      <Provider store={store}>
        <BrowserRouter>
          <ModalSuccess />
        </BrowserRouter>
      </Provider>
    );
  };

  test('Should render the component', async () => {
    const {asFragment} = setup();
    
    await act(() => Promise.resolve())

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText(/Payment Succesful/i)).toBeDefined();
    expect(screen.getByText(/You will receive your order in the coming days/i)).toBeDefined();
  });

  test('Should dispatches action to close modal when Ok button is clicked', () => {
    setup();

    // Simula clic en el botón 'Ok'
    fireEvent.click(screen.getByRole('button', { name: /Ok/i }));

    // Verifica que se despache la acción para cerrar el modal
    const actions = store.getActions();
    expect(actions).toEqual([setModalSuccessSlice(false)]);
  });
});
