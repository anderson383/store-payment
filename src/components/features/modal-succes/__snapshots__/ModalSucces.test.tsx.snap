// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in ModalSuccess component Should render the component 1`] = `<DocumentFragment />`;
